presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -166,19 +170,26 @@ private void validateShowStatsSubquery(ShowStats node, Query query, QuerySpecifi
 
         private Node rewriteShowStats(ShowStats node, Table table, List<SelectItem> selectItems, Constraint constraint)
         {
-            List<String> statsColumnNames = buildColumnsNames();
             TableHandle tableHandle = getTableHandle(node, table.getName());
             TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, constraint);
             TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
 
-            Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
             Multimap<String, String> extractNamedResultColumns = extractNamedResultColumns(node, tableMetadata, selectItems);
+
+            try {
+                accessControl.checkCanSelectFromColumns(session.toSecurityContext(), tableMetadata.getQualifiedName(), extractNamedResultColumns.keySet());
+            }
+            catch (AccessDeniedException e) {
+                throw new AccessDeniedException(format("Cannot show stats for table %s", tableMetadata.getTable().toString()));
+            }
+
+            Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
             List<Expression> resultRows = buildStatisticsRows(tableMetadata, columnHandles, extractNamedResultColumns, tableStatistics);
 
-            return simpleQuery(selectAll(buildSelectItems(statsColumnNames)),
+            return simpleQuery(selectAll(buildSelectItems()),
                     aliased(new Values(resultRows),
                             "table_stats_for_" + table.getName(),
-                            statsColumnNames));
+                            buildColumnsNames()));
```

Comment:
unnecessary change?

Reply:
buildColumnsNames() are used in this single place. That's why I've inlined it